starknet_py/net/client_models.py

```diff
@@ -253,20 +257,13 @@ class TransactionReceipt:
     block_number: Optional[int] = None
     block_hash: Optional[int] = None
     actual_fee: int = 0
-    # TODO (#1047): change that into ExecutionResources class after gateway removal
-    #  (values of course differ for each client)
     # TODO (#1179): this field should be required
-    execution_resources: Optional[dict] = field(default_factory=dict)
+    execution_resources: Optional[ExecutionResources] = None
 
     message_hash: Optional[int] = None  # L1_HANDLER_TXN_RECEIPT-only
 
     rejection_reason: Optional[str] = None
```

Comment:
Since we are removing gateway-only `revert_error`, maybe remove `rejection_reason` as well, or do we have a reason to keep it?

Reply:
Same response as [here](https://github.com/software-mansion/starknet.py/pull/1226/files#r1405971792)